kafka-admin/src/main/java/io/strimzi/admin/kafka/admin/KafkaAdminService.java

```diff
@@ -38,22 +50,28 @@
 
                 final TypeDefinitionRegistry typeDefinitionRegistry = schemaParser.parse(userInputStream);
                 p.complete(typeDefinitionRegistry);
-            }
-            catch (Exception exc) {
+            } catch (Exception exc) {
                 p.fail(exc);
             }
         }, ar -> {
-            final TypeDefinitionRegistry schema = (TypeDefinitionRegistry) ar.result();
+                final TypeDefinitionRegistry schema = (TypeDefinitionRegistry) ar.result();
 
-            final RuntimeWiring query = RuntimeWiring.newRuntimeWiring()
-                .type("Query", typeWiring -> typeWiring
-                    .dataFetcher("topic", new VertxDataFetcher<>(TopicHandler::getTopic))
-                    .dataFetcher("topicList", new VertxDataFetcher<>(TopicListHandler::getTopicList))
-                )
-                .build();
+                AdminClientProvider acp = new AdminClientProvider(vertx, config);
+                acp.open();
```

Comment:
if the Kafka cluster is not ready, it's running on the Vert.x main thread blocking it. We should avoid that doing asynchronously.